backend/routes/events.py

```
@@ -0,0 +1,63 @@
+    current_user: User = Depends(get_current_user),
+    db: Session = Depends(get_db)
+):
+    # Safely slice the event IDs list
+    paged_ids = event_ids[offset : offset + limit]
+
+    if not paged_ids:
+        return []
+
+    events = db.query(Event).filter(Event.id.in_(paged_ids)).all()
+
+    return events
```

Ruff (0.8.2)

Unnecessary assignment to `events` before `return` statement

Remove unnecessary assignment

(RET504)